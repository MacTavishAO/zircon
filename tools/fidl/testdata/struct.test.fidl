library fidl.test.json;

struct Simple {
    uint8 f1;
    bool f2;
};

struct BasicStruct {
    uint32 x;
    string y;
};
