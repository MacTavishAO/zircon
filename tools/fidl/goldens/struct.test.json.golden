{
  "version": "0.0.1",
  "name": "fidl.test.json",
  "library_dependencies": [],
  "bits_declarations": [],
  "const_declarations": [],
  "enum_declarations": [],
  "experimental_resource_declarations": [],
  "interface_declarations": [],
  "service_declarations": [],
  "struct_declarations": [
    {
      "name": "fidl.test.json/Simple",
      "location": {
        "filename": "struct.test.fidl",
        "line": 3,
        "column": 8,
        "length": 6
      },
      "anonymous": false,
      "members": [
        {
          "type": {
            "kind": "primitive",
            "subtype": "uint8"
          },
          "name": "f1",
          "location": {
            "filename": "struct.test.fidl",
            "line": 4,
            "column": 11,
            "length": 2
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 0
          }
        },
        {
          "type": {
            "kind": "primitive",
            "subtype": "bool"
          },
          "name": "f2",
          "location": {
            "filename": "struct.test.fidl",
            "line": 5,
            "column": 10,
            "length": 2
          },
          "field_shape_v1": {
            "offset": 1,
            "padding": 0
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 2,
        "alignment": 1,
        "depth": 0,
        "max_handles": 0,
        "max_out_of_line": 0,
        "has_padding": false,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/BasicStruct",
      "location": {
        "filename": "struct.test.fidl",
        "line": 8,
        "column": 8,
        "length": 11
      },
      "anonymous": false,
      "members": [
        {
          "type": {
            "kind": "primitive",
            "subtype": "uint32"
          },
          "name": "x",
          "location": {
            "filename": "struct.test.fidl",
            "line": 9,
            "column": 12,
            "length": 1
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 4
          }
        },
        {
          "type": {
            "kind": "string",
            "nullable": false
          },
          "name": "y",
          "location": {
            "filename": "struct.test.fidl",
            "line": 10,
            "column": 12,
            "length": 1
          },
          "field_shape_v1": {
            "offset": 8,
            "padding": 0
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 24,
        "alignment": 8,
        "depth": 1,
        "max_handles": 0,
        "max_out_of_line": 4294967295,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    }
  ],
  "table_declarations": [],
  "union_declarations": [],
  "type_alias_declarations": [],
  "declaration_order": [
    "fidl.test.json/Simple",
    "fidl.test.json/BasicStruct"
  ],
  "declarations": {
    "fidl.test.json/Simple": "struct",
    "fidl.test.json/BasicStruct": "struct"
  }
}
